analysis/analysis_utils.py

```diff
@@ -238,53 +238,60 @@ def save_results(self, res_filepath, out_suffix, keep):
+            within the image.
+
+        Note
+        ----
+        It is not neccessary to calculate this angle for each image in an
+        image set if they have all been warped to a common WCS.
+
+        See Also
+        --------
+        run_search.do_gpu_search
         """
-        pixel_coords = [[], []]
-        pixel_start = [[1000, 2000]]
-        angle = float(angle_to_ecliptic)
-        vel_array = np.array([[6.0 * np.cos(angle), 6.0 * np.sin(angle)]])
-        time_array = [0.0, 1.0, 2.0]
-        vel_par_arr = vel_array[:, 0]
-        vel_perp_arr = vel_array[:, 1]
-
-        if type(vel_par_arr) is not np.ndarray:
-            vel_par_arr = [vel_par_arr]
-        if type(vel_perp_arr) is not np.ndarray:
-            vel_perp_arr = [vel_perp_arr]
-        for start_loc, vel_par, vel_perp in zip(pixel_start, vel_par_arr, vel_perp_arr):
-
-            start_coord = astroCoords.SkyCoord.from_pixel(start_loc[0], start_loc[1], test_wcs)
-            eclip_coord = start_coord.geocentrictrueecliptic
-            eclip_l = []
-            eclip_b = []
-            for time_step in time_array:
-                eclip_l.append(eclip_coord.lon + vel_par * time_step * u.arcsec)
-                eclip_b.append(eclip_coord.lat + vel_perp * time_step * u.arcsec)
-            eclip_vector = astroCoords.SkyCoord(eclip_l, eclip_b, frame="geocentrictrueecliptic")
-            pixel_coords_set = astroCoords.SkyCoord.to_pixel(eclip_vector, test_wcs)
-            pixel_coords[0].append(pixel_coords_set[0])
-            pixel_coords[1].append(pixel_coords_set[1])
-
-        pixel_coords = np.array(pixel_coords)
-        x_dist = pixel_coords[0, 0, -1] - pixel_coords[0, 0, 0]
-        y_dist = pixel_coords[1, 0, -1] - pixel_coords[1, 0, 0]
-        eclip_angle = np.arctan2(y_dist, x_dist)
-
-        return eclip_angle
+        # pick a starting pixel approximately near the center of the image
+        # convert it to ecliptic coordinates
+        start_pixel = np.array([1000, 2000])
+        start_pixel_coord = astroCoords.SkyCoord.from_pixel(
+            start_pixel[0], 
+            start_pixel[1], 
+            wcs)
+        start_ecliptic_coord = start_pixel_coord.geocentrictrueecliptic
+
+        # pick a guess pixel by moving parallel to the ecliptic
+        # convert it to pixel coordinates for the given WCS
+        vel_par = 6
+        vel_perp = 0
+        time_step = 2
+        guess_ecliptic_coord = astroCoords.SkyCoord(
+            start_ecliptic_coord.lon + vel_par*time_step*u.arcsec,
+            start_ecliptic_coord.lat + vel_perp*time_step*u.arcsec,
+            frame="geocentrictrueecliptic")
+        guess_pixel_coord = guess_ecliptic_coord.to_pixel(wcs)
+        
+        # calculate the distance, in pixel coordinates, between the guess and 
+        # the start pixel. Calculate the angle that represents in the image.
+        x_dist, y_dist = np.array(guess_pixel_coord) - start_pixel
+        return np.arctan2(y_dist/x_dist) 
```

Comment:
arctan2 takes the x and y parameters separately.

Reply:
Ah, I was in a rush, there's also a bug above where I don't move parallel with the ecliptic, but move both latitude and longitude. Nice catch.